<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="enabled">
   <bool>true</bool>
  </property>
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1080</width>
    <height>720</height>
   </rect>
  </property>
  <property name="minimumSize">
   <size>
    <width>1080</width>
    <height>720</height>
   </size>
  </property>
  <property name="maximumSize">
   <size>
    <width>1080</width>
    <height>720</height>
   </size>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <widget class="QTabWidget" name="Graphs">
    <property name="geometry">
     <rect>
      <x>5</x>
      <y>8</y>
      <width>560</width>
      <height>660</height>
     </rect>
    </property>
    <property name="minimumSize">
     <size>
      <width>560</width>
      <height>660</height>
     </size>
    </property>
    <property name="maximumSize">
     <size>
      <width>560</width>
      <height>660</height>
     </size>
    </property>
    <property name="currentIndex">
     <number>4</number>
    </property>
    <widget class="QWidget" name="reExps">
     <attribute name="title">
      <string>RE exps</string>
     </attribute>
     <widget class="QWidget" name="verticalLayoutWidget">
      <property name="geometry">
       <rect>
        <x>-1</x>
        <y>9</y>
        <width>551</width>
        <height>621</height>
       </rect>
      </property>
      <layout class="QVBoxLayout" name="verticalLayout">
       <item>
        <widget class="QTextBrowser" name="reRules"/>
       </item>
      </layout>
     </widget>
    </widget>
    <widget class="QWidget" name="nfa">
     <attribute name="title">
      <string>nfa</string>
     </attribute>
     <widget class="QWidget" name="verticalLayoutWidget_6">
      <property name="geometry">
       <rect>
        <x>0</x>
        <y>170</y>
        <width>551</width>
        <height>461</height>
       </rect>
      </property>
      <layout class="QVBoxLayout" name="nfaLayout">
       <item>
        <widget class="QTableView" name="nfaTab"/>
       </item>
      </layout>
     </widget>
     <widget class="QWidget" name="nfaMenu" native="true">
      <property name="geometry">
       <rect>
        <x>0</x>
        <y>10</y>
        <width>551</width>
        <height>151</height>
       </rect>
      </property>
      <widget class="QComboBox" name="nfaSel">
       <property name="geometry">
        <rect>
         <x>220</x>
         <y>30</y>
         <width>281</width>
         <height>31</height>
        </rect>
       </property>
       <property name="font">
        <font>
         <family>Adobe Arabic</family>
         <pointsize>14</pointsize>
        </font>
       </property>
       <item>
        <property name="text">
         <string>None</string>
        </property>
       </item>
      </widget>
      <widget class="QLabel" name="label">
       <property name="geometry">
        <rect>
         <x>50</x>
         <y>30</y>
         <width>161</width>
         <height>31</height>
        </rect>
       </property>
       <property name="font">
        <font>
         <family>Adobe Arabic</family>
         <pointsize>20</pointsize>
        </font>
       </property>
       <property name="text">
        <string>NFA selection:</string>
       </property>
      </widget>
     </widget>
    </widget>
    <widget class="QWidget" name="dfa">
     <attribute name="title">
      <string>dfa</string>
     </attribute>
     <widget class="QWidget" name="verticalLayoutWidget_7">
      <property name="geometry">
       <rect>
        <x>0</x>
        <y>170</y>
        <width>551</width>
        <height>461</height>
       </rect>
      </property>
      <layout class="QVBoxLayout" name="dfaLayout">
       <item>
        <widget class="QTableView" name="dfaTab"/>
       </item>
      </layout>
     </widget>
     <widget class="QWidget" name="dfaMenu" native="true">
      <property name="geometry">
       <rect>
        <x>0</x>
        <y>10</y>
        <width>551</width>
        <height>151</height>
       </rect>
      </property>
      <widget class="QComboBox" name="dfaSel">
       <property name="geometry">
        <rect>
         <x>220</x>
         <y>30</y>
         <width>281</width>
         <height>31</height>
        </rect>
       </property>
       <property name="font">
        <font>
         <family>Adobe Arabic</family>
         <pointsize>14</pointsize>
        </font>
       </property>
       <item>
        <property name="text">
         <string>None</string>
        </property>
       </item>
      </widget>
      <widget class="QLabel" name="label_2">
       <property name="geometry">
        <rect>
         <x>50</x>
         <y>30</y>
         <width>161</width>
         <height>31</height>
        </rect>
       </property>
       <property name="font">
        <font>
         <family>Adobe Arabic</family>
         <pointsize>20</pointsize>
        </font>
       </property>
       <property name="text">
        <string>DFA selection:</string>
       </property>
      </widget>
     </widget>
    </widget>
    <widget class="QWidget" name="minDfa">
     <attribute name="title">
      <string>minDfa</string>
     </attribute>
     <widget class="QWidget" name="verticalLayoutWidget_8">
      <property name="geometry">
       <rect>
        <x>0</x>
        <y>170</y>
        <width>551</width>
        <height>461</height>
       </rect>
      </property>
      <layout class="QVBoxLayout" name="minDfaLayout">
       <item>
        <widget class="QTableView" name="minDfaTab"/>
       </item>
      </layout>
     </widget>
     <widget class="QFrame" name="frame">
      <property name="geometry">
       <rect>
        <x>0</x>
        <y>10</y>
        <width>551</width>
        <height>151</height>
       </rect>
      </property>
      <property name="frameShape">
       <enum>QFrame::StyledPanel</enum>
      </property>
      <property name="frameShadow">
       <enum>QFrame::Raised</enum>
      </property>
      <widget class="QLabel" name="label_3">
       <property name="geometry">
        <rect>
         <x>30</x>
         <y>30</y>
         <width>181</width>
         <height>31</height>
        </rect>
       </property>
       <property name="font">
        <font>
         <family>Adobe Arabic</family>
         <pointsize>20</pointsize>
        </font>
       </property>
       <property name="text">
        <string>minDFA selection:</string>
       </property>
      </widget>
      <widget class="QComboBox" name="minDfaSel">
       <property name="geometry">
        <rect>
         <x>220</x>
         <y>30</y>
         <width>281</width>
         <height>31</height>
        </rect>
       </property>
       <property name="font">
        <font>
         <family>Adobe Arabic</family>
         <pointsize>14</pointsize>
        </font>
       </property>
       <item>
        <property name="text">
         <string>None</string>
        </property>
       </item>
      </widget>
     </widget>
    </widget>
    <widget class="QWidget" name="cppCode">
     <attribute name="title">
      <string>.cpp</string>
     </attribute>
     <widget class="QWidget" name="verticalLayoutWidget_2">
      <property name="geometry">
       <rect>
        <x>0</x>
        <y>10</y>
        <width>551</width>
        <height>621</height>
       </rect>
      </property>
      <layout class="QVBoxLayout" name="verticalLayout_2">
       <item>
        <widget class="QTextBrowser" name="cppText"/>
       </item>
      </layout>
     </widget>
    </widget>
    <widget class="QWidget" name="tnyFile">
     <attribute name="title">
      <string>.tny</string>
     </attribute>
     <widget class="QWidget" name="verticalLayoutWidget_3">
      <property name="geometry">
       <rect>
        <x>0</x>
        <y>10</y>
        <width>551</width>
        <height>621</height>
       </rect>
      </property>
      <layout class="QVBoxLayout" name="verticalLayout_3">
       <item>
        <widget class="QTextBrowser" name="tnyCode"/>
       </item>
      </layout>
     </widget>
    </widget>
    <widget class="QWidget" name="lexFile">
     <attribute name="title">
      <string>.lex</string>
     </attribute>
     <widget class="QWidget" name="verticalLayoutWidget_4">
      <property name="geometry">
       <rect>
        <x>0</x>
        <y>10</y>
        <width>551</width>
        <height>621</height>
       </rect>
      </property>
      <layout class="QVBoxLayout" name="verticalLayout_4">
       <item>
        <widget class="QTextBrowser" name="lexRes"/>
       </item>
      </layout>
     </widget>
    </widget>
   </widget>
   <widget class="QWidget" name="verticalLayoutWidget_5">
    <property name="geometry">
     <rect>
      <x>600</x>
      <y>30</y>
      <width>461</width>
      <height>631</height>
     </rect>
    </property>
    <layout class="QVBoxLayout" name="verticalLayout_5">
     <item>
      <widget class="QWidget" name="widget" native="true">
       <widget class="QGroupBox" name="terminal">
        <property name="geometry">
         <rect>
          <x>-1</x>
          <y>-1</y>
          <width>461</width>
          <height>311</height>
         </rect>
        </property>
        <property name="title">
         <string>Terminal</string>
        </property>
        <widget class="QTextBrowser" name="terLines">
         <property name="geometry">
          <rect>
           <x>0</x>
           <y>20</y>
           <width>460</width>
           <height>280</height>
          </rect>
         </property>
         <property name="minimumSize">
          <size>
           <width>460</width>
           <height>280</height>
          </size>
         </property>
         <property name="maximumSize">
          <size>
           <width>460</width>
           <height>280</height>
          </size>
         </property>
        </widget>
       </widget>
      </widget>
     </item>
     <item>
      <widget class="QWidget" name="widget_2" native="true">
       <widget class="QWidget" name="widget_3" native="true">
        <property name="geometry">
         <rect>
          <x>0</x>
          <y>0</y>
          <width>461</width>
          <height>311</height>
         </rect>
        </property>
        <widget class="QGroupBox" name="Menu">
         <property name="geometry">
          <rect>
           <x>0</x>
           <y>0</y>
           <width>461</width>
           <height>311</height>
          </rect>
         </property>
         <property name="title">
          <string>Tool Menu</string>
         </property>
         <widget class="QWidget" name="widget_4" native="true">
          <property name="geometry">
           <rect>
            <x>10</x>
            <y>30</y>
            <width>441</width>
            <height>161</height>
           </rect>
          </property>
          <widget class="QWidget" name="gridLayoutWidget">
           <property name="geometry">
            <rect>
             <x>10</x>
             <y>0</y>
             <width>411</width>
             <height>161</height>
            </rect>
           </property>
           <layout class="QGridLayout" name="gridLayout">
            <item row="0" column="0">
             <widget class="QWidget" name="widget_5" native="true">
              <widget class="QPushButton" name="transNFA">
               <property name="geometry">
                <rect>
                 <x>40</x>
                 <y>40</y>
                 <width>111</width>
                 <height>28</height>
                </rect>
               </property>
               <property name="text">
                <string>RE -&gt; NFA</string>
               </property>
              </widget>
              <widget class="QPushButton" name="minimizeDFA">
               <property name="geometry">
                <rect>
                 <x>40</x>
                 <y>100</y>
                 <width>111</width>
                 <height>28</height>
                </rect>
               </property>
               <property name="text">
                <string>minimize DFA</string>
               </property>
              </widget>
             </widget>
            </item>
            <item row="0" column="1">
             <widget class="QWidget" name="widget_6" native="true">
              <widget class="QPushButton" name="transDFA">
               <property name="geometry">
                <rect>
                 <x>40</x>
                 <y>40</y>
                 <width>111</width>
                 <height>28</height>
                </rect>
               </property>
               <property name="text">
                <string>NFA -&gt; DFA</string>
               </property>
              </widget>
              <widget class="QPushButton" name="produceCPP">
               <property name="geometry">
                <rect>
                 <x>40</x>
                 <y>100</y>
                 <width>111</width>
                 <height>28</height>
                </rect>
               </property>
               <property name="text">
                <string>Code produce</string>
               </property>
              </widget>
             </widget>
            </item>
           </layout>
          </widget>
         </widget>
         <widget class="QPushButton" name="saveLex">
          <property name="geometry">
           <rect>
            <x>270</x>
            <y>210</y>
            <width>111</width>
            <height>28</height>
           </rect>
          </property>
          <property name="text">
           <string>Display .lex</string>
          </property>
         </widget>
         <widget class="QPushButton" name="clearText">
          <property name="geometry">
           <rect>
            <x>60</x>
            <y>210</y>
            <width>111</width>
            <height>28</height>
           </rect>
          </property>
          <property name="text">
           <string>Clear</string>
          </property>
         </widget>
        </widget>
       </widget>
      </widget>
     </item>
    </layout>
   </widget>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>1080</width>
     <height>26</height>
    </rect>
   </property>
   <widget class="QMenu" name="menuFile">
    <property name="title">
     <string>File</string>
    </property>
    <addaction name="openFile"/>
   </widget>
   <addaction name="menuFile"/>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
  <action name="openFile">
   <property name="text">
    <string>open file</string>
   </property>
  </action>
 </widget>
 <resources/>
 <connections/>
</ui>
